src/AsmResolver.DotNet/Signatures/GenericContext.cs

```diff
@@ -62,14 +69,111 @@ public TypeSignature GetTypeArgument(GenericParameterSignature parameter)
                 GenericParameterType.Method => Method,
                 _ => throw new ArgumentOutOfRangeException()
             };
-            
+
             if (argumentProvider is null)
-                throw new ArgumentOutOfRangeException();
+                return parameter;
 
             if (parameter.Index >= 0 && parameter.Index < argumentProvider.TypeArguments.Count)
                 return argumentProvider.TypeArguments[parameter.Index];
 
             throw new ArgumentOutOfRangeException();
         }
+
+
+        /// <summary>
+        /// Gets type generic context from <see cref="TypeSpecification"/>.
+        /// </summary>
+        /// <param name="type">Type specification to get generic context from.</param>
```

Comment:
Type specification to get **the** generic context from.

Applies to all `FromXXX` methods.